src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -52,10 +58,56 @@ class RegularSync(
       askForHeaders()
   }
 
-  def running: Receive = handleCommonMessages orElse {
-    case ResumeRegularSync =>
-      askForHeaders()
+  private def resumeRegularSync(): Unit = {
+    resumeRegularSyncTimeout.foreach(_.cancel)
+    resumeRegularSyncTimeout = None
+    askForHeaders()
+  }
 
+  /**
+    * Handles broadcasted blocks, processing them and inserting them to the blockchain if necessary
+    * FIXME: Current implementation is naive in that NewBlock msgs that can't be immediately processed are discarded
+    *        and requested again to the peers
+    */
+  def handleBroadcastedBlockMessages: Receive = {
+    case MessageFromPeer(NewBlock(newBlock, _), peerId) =>
+      //we allow inclusion of new block only if we are not syncing / reorganising chain
```

Comment:
What exactly does it mean? We asked for a blocks and waiting for response, right? Any other circumstances? What would happen if we ignored that? 

I'm asking because I'm still not sure as to what logic exactly should be implemented in [Ledger.importBlock](https://github.com/input-output-hk/mantis/pull/305/files#diff-1b4e47475b368c25eb5141199a4289d0R34). Ideally we should be able to just throw any blocks at it without a care in the world.

Reply:
I also think that ideally we should do so. However, due to the refactoring being done next, I would leave the current implementation as is, as removing this condition could have unexpected results as it could cause our client to ask for headers, causing it to possibly have two pending requests at once (if any `resumeRegularSync()` is called).

>What exactly does it mean? We asked for some blocks and we're waiting for response, right? Any other circumstances? What would happen if we ignored that?

That's right, IMO this only includes when the node is syncing and when it periodically asks it's peers for the latest blocks (due to [resumeRegularSyncTimeout](https://github.com/input-output-hk/mantis/pull/302/files#diff-65b863fb327f34212f7a48e87ccdec13R43)).

Reply:
Sure, no changes required here, I was just asking in relation to my task.